applications/nrf5340_audio/src/audio/audio_datapath.c

```diff
@@ -792,9 +798,31 @@ void audio_datapath_sdu_ref_update(uint32_t sdu_ref_us)
 	}
 }
 
-void audio_datapath_stream_out(const uint8_t *buf, size_t size, uint32_t sdu_ref_us, bool bad_frame,
-			       uint32_t recv_frame_ts_us)
+void audio_datapath_stream_out(const uint8_t *buf, size_t size, uint32_t sdu_ref_us,
+			       uint16_t seq_num, bool bad_frame, uint32_t recv_frame_ts_us)
 {
+	static uint32_t last_time_stamp;
+	uint32_t delta_time_stamp = 0;
+	/* Updates compare value for setting GPIOTE Sync LED */
+	if ((seq_num % SYNC_LED_SEQ_NR_PERIOD) == SYNC_LED_SEQ_NR_ON) {
+		uint32_t current_time_stamp = audio_sync_timer_curr_time_get();
+		delta_time_stamp = current_time_stamp - last_time_stamp;
+
+		uint32_t new_compare_set_time = current_time_stamp + delta_time_stamp;
+		audio_sync_timer_led_on_cmpr_time_set(new_compare_set_time);
+		last_time_stamp = current_time_stamp;
+	}
+
+	/* Updates compare value for clearing GPIOTE Sync LED */
+	else if ((seq_num % SYNC_LED_SEQ_NR_PERIOD) == SYNC_LED_SEQ_NR_OFF) {
```

Comment:
It should be:
```
if (statement) {
    /* Do something */
} else if (next_statement) {
    /* Do something else */
}
```
There shouldn't be blank lines in between

Reply:
Also, this code is identical except for the function call, so it could be rewritten. Suggestion, remove if and else if and replace with:
```suggestion
	if ((seq_num % SYNC_LED_SEQ_NR_PERIOD) == SYNC_LED_SEQ_NR_ON ||
	     seq_num % SYNC_LED_SEQ_NR_PERIOD) == SYNC_LED_SEQ_NR_OFF) {
	        uint32_t current_time_stamp = audio_sync_timer_curr_time_get();
		uint32_t new_compare_time =  current_time_stamp + (current_time_stamp - last_time_stamp);
		bool clear = (seq_num % SYNC_LED_SEQ_NR_PERIOD == SYNC_LED_SEQ_NR_OFF ? true : false);

		audio_sync_timer_led_cmpr_time_set(new_compare_time, clear);
		last_time_stamp = current_time_stamp;
	}
```

Reply:
Does this need to take presentation delay into account?

Reply:
Currently the LEDS are blinking synchronized, but the plan is to check the accuracy with an logic analyzer.

Reply:
We need to at least use sdu_ref_us somewhere in these calculations

Reply:
Added a suggestion with sdu_ref_us